### PR TITLE
DMP-502 Created a script to create the external_object_directory table

### PR DESCRIPTION
**Pull request by marcmollins**
…e entity class to go along with it

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DMP-502

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    public static final String EXTERNAL_LOCATION = "external_location";
+    public static final String EXTERNAL_LOCATION_TYPE = "external_location_type";
+    public static final String CREATED_DATE_TIME = "created_ts";
+    public static final String LAST_UPDATED_DATE_TIME = "modified_ts";
+    public static final String MODIFIED_BY = "modified_by";
+    public static final String STATUS = "status";
+    public static final String CHECKSUM = "checksum";
+    public static final String ATTEMPTS = "attempts";
+    public static final String TABLE_NAME = "external_object_directory";
+
```
Comment (tonymort) [inline src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java:35]: Do these need to be constants?  Are they accessed outside of ExternalObjectDocumentDirectory?

File: src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@Entity
+@Table(name = ExternalObjectDocumentDirectory.TABLE_NAME)
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ExternalObjectDocumentDirectory {
+
+    public static final String ID = "eod_id";
+    public static final String MEDIA_ID = "moj_media_id";
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java:23]: Should this be `moj_med_id`?

File: src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    public static final String TRANSCRIPTION_ID = "moj_tra_id";
+    public static final String ANNOTATION_ID = "moj_ann_id";
+    public static final String EXTERNAL_LOCATION = "external_location";
+    public static final String EXTERNAL_LOCATION_TYPE = "external_location_type";
```
Comment (richardbull) [inline src/main/java/uk/gov/hmcts/darts/common/entity/ExternalObjectDocumentDirectory.java:27]: Please check with @hemantasharma1129  - whether he expects us to be adding a "xxx_type" table as well?

File: src/main/resources/db/migration/V1_12__create_external_object_document_directory_table.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  eod_id                         INTEGER                     NOT NULL
+, moj_med_id                     INTEGER
+, moj_tra_id                     INTEGER
+, moj_ann_id                     INTEGER
+, external_location              CHARACTER VARYING           NOT NULL
+, external_location_type         CHARACTER VARYING           NOT NULL
+, created_ts                     TIMESTAMP WITH TIME ZONE    NOT NULL
+, modified_ts                    TIMESTAMP WITH TIME ZONE    NOT NULL
+, modified_by                    INTEGER                     NOT NULL
+, status                         INTEGER                     NOT NULL
```
Comment (richardbull) [inline src/main/resources/db/migration/V1_12__create_external_object_document_directory_table.sql:11]: Please check with @hemantasharma1129  - whether he expects us to be adding a "xxx_status_type" table as well?

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**